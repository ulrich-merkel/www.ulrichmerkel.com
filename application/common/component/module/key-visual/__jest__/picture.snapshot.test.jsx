/* eslint-disable import/no-extraneous-dependencies, func-names */
import React from 'react';
import renderer from 'react-test-renderer';
import ModuleKeyVisualPicture from './../picture';

describe('common/component/module/key-visual/picture', function () {
    const defaultProps = {
        img: {
            name: 'keyvisual',
            ext: 'jpg',
            path: '/img/content/home/',
            alt: 'foo',
            sizes: [
                {
                    width: 2560,
                    height: 1600,
                    minWidth: 2560
                },
                {
                    width: 1680,
                    height: 1050,
                    minWidth: 1680
                },
                {
                    width: 1440,
                    height: 960,
                    minWidth: 1440
                }
            ]
        }
    };

    it('should render correctly', function () {
        const tree = renderer.create(
            <ModuleKeyVisualPicture
                {...defaultProps}
            />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('should render null if no content is provided', function () {
        const tree = renderer.create(
            <ModuleKeyVisualPicture
                {...defaultProps}
                img={{
                    alt: 'foo',
                    sizes: [
                        {
                            width: 2560,
                            height: 1600,
                            minWidth: 2560
                        },
                        {
                            width: 1680,
                            height: 1050,
                            minWidth: 1680
                        },
                        {
                            width: 1440,
                            height: 960,
                            minWidth: 1440
                        }
                    ]
                }}
            />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('should render type and isCovered if given', function () {
        const tree = renderer.create(
            <ModuleKeyVisualPicture
                {...defaultProps}
                type={'digital'}
                isCovered
            />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
