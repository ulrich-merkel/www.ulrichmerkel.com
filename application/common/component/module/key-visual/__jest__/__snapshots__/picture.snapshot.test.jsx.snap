// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/component/module/key-visual/picture should render correctly 1`] = `
<picture
  className="c-picture m-key-visual__image m-key-visual__image--digital"
  itemScope={true}
  itemType="http://schema.org/ImageObject"
  style={Object {}}
>
  <source
    className="c-picture__source"
    media="(min-width: 2560px)"
    srcSet="/img/content/home/keyvisual@2560x1600.jpg"
  />
  <source
    className="c-picture__source"
    media="(min-width: 1680px)"
    srcSet="/img/content/home/keyvisual@1680x1050.jpg"
  />
  <source
    className="c-picture__source"
    media="(min-width: 1440px)"
    srcSet="/img/content/home/keyvisual@1440x960.jpg"
  />
  <img
    alt="foo"
    className="c-picture__img"
    itemProp="contentUrl"
    src="data:image/gifbase64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
    srcSet="/img/content/home/keyvisual.jpg"
  />
</picture>
`;

exports[`common/component/module/key-visual/picture should render null if no content is provided 1`] = `null`;

exports[`common/component/module/key-visual/picture should render type and isCovered if given 1`] = `
<picture
  className="c-picture m-key-visual__image m-key-visual__image--digital"
  itemScope={true}
  itemType="http://schema.org/ImageObject"
  style={Object {}}
>
  <source
    className="c-picture__source"
    media="(min-width: 2560px)"
    srcSet="/img/content/home/keyvisual@2560x1600.jpg"
  />
  <source
    className="c-picture__source"
    media="(min-width: 1680px)"
    srcSet="/img/content/home/keyvisual@1680x1050.jpg"
  />
  <source
    className="c-picture__source"
    media="(min-width: 1440px)"
    srcSet="/img/content/home/keyvisual@1440x960.jpg"
  />
  <img
    alt="foo"
    className="c-picture__img"
    itemProp="contentUrl"
    src="data:image/gifbase64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
    srcSet="/img/content/home/keyvisual.jpg"
  />
</picture>
`;
